### PR TITLE
#2453: partner oauth2 login fixes

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Part of #2453
- Fixes: https://github.com/pixiebrix/pixiebrix-extension/issues/4594
- Fixes: https://github.com/pixiebrix/pixiebrix-extension/issues/4628
- Add `controlRoomUrl` field to `managedStorageSchema.json` for enterprise IT departments to set as part of enterprise rollout
- PartnerSetupCard: Check and use configured `controlRoomUrl` in managed storage
- ControlRoomOAuthForm: fix `null` value handling for `settingsSlice.authServiceId`
- useRequiredPartnerAuth: check for  configured `controlRoomUrl` in managed storage, fix partner JWT check

## Remaining Work

- [x] Refresh AuthState from /api/me endpoint after OAuth2 login
- [x] When going round trip -- back to Extension Console from Admin Console, showing Login with AA screen. And every visit to Admin Console shows "Successfully linked the Browser Extension to your PixieBrix account"

## Demo

- [x] Loom demo: https://www.loom.com/share/2456bffe6c2842979700b9d8532a23e6

## Future Work / Out of Scope

- Check actual Control Room URL in cached partner auth data matches expected Control Room URL. (Unlikely for now that enterprise rollout users would need to switch)

## Team Coordination

- [X] This PR requires a documentation change (link to old docs): update documentation: https://docs.pixiebrix.com/integrations/automation-anywhere/aari-extensions-enterprise-it-guide#ddba78e8db8042afb230058e98ec563c

## Checklist

- [x] Add tests
- [x] Run Storybook and manually confirm that all stories are working
- [x] Designate a primary reviewer: @johnnymetz

### CODE REVIEW

File: src/options/pages/onboarding/SetupPage.test.tsx [hunk 1]
```diff
@@ -106,4 +111,77 @@ describe("SetupPage", () => {
...
+    // User will be unauthenticated
+    (useGetMeQuery as jest.Mock).mockImplementation(() => ({
+      isLoading: false,
+      data: {},
+    }));
+
+    (global as any).window = Object.create(window);
+
+    const url = new URL(
+      "chrome-extension://abc/options.html?hostname=https://mycontrolroom.com#/start"
```
Comment (twschiller) [inline src/options/pages/onboarding/SetupPage.test.tsx:125]: - [ ] @mnholtz can you confirm this is the URL format? In particular, that the hostname is a search param on the main URL, not part of the hash?

File: src/options/pages/onboarding/partner/ControlRoomOAuthForm.tsx [hunk 2]
```diff
@@ -40,7 +41,8 @@ export type ControlRoomConfiguration = {
 };
 
 const validationSchema = Yup.object().shape({
-  controlRoomUrl: Yup.string().url().required(),
+  // Leave off .url() because it doesn't allow localhost (which is needed for using mock-server for testing)
```
Comment (johnnymetz) [inline src/options/pages/onboarding/partner/ControlRoomOAuthForm.tsx:44]: nit: we can remove `.url()` in dev mode but include it in prod, see [here](https://github.com/pixiebrix/pixiebrix-app/pull/2562/files#diff-12f266ec8e7984779e35782ba567250bc5d9c18d8e03bc181f89686c95dcee65R23-R28)

File: src/services/api.ts [hunk 3]
```diff
@@ -469,4 +469,5 @@ export const {
   useListPackageVersionsQuery,
   useUpdateScopeMutation,
   useGetStarterBlueprintsQuery,
+  util,
```
Comment (johnnymetz) [inline src/services/api.ts:472]: nit: unused import

**Review by johnnymetz — APPROVED** [review]

File: src/services/api.ts (hunk 3 above)
Comment (twschiller) [reply src/services/api.ts:472]: It's used here: https://github.com/pixiebrix/pixiebrix-extension/pull/4627/files#diff-f32cb3f188e731cc4652c4102319ef1a98d6d96056d691a9cc6ceb7bd200aac0R36

File: src/options/Navbar.tsx [hunk 4]
```diff
@@ -17,42 +17,70 @@
...
 import { toggleSidebar } from "./toggleSidebar";
 import { SettingsState } from "@/store/settingsTypes";
 import cx from "classnames";
 import { selectAuth } from "@/auth/authSelectors";
 import { ThemeLogo } from "@/utils/themeUtils";
+import useLinkState from "@/auth/useLinkState";
+
+function useAdminConsoleUrl(): string {
+  // Need to update serviceURL on changes to partner auth data:
+  // https://github.com/pixiebrix/pixiebrix-extension/issues/4594
```
Comment (johnnymetz) [inline src/options/Navbar.tsx:42]: Look like this is done so can delete this comment

**Review by johnnymetz — APPROVED** [review]

File: src/options/pages/onboarding/SetupPage.test.tsx (hunk 1 above)
Comment (mnholtz) [reply src/options/pages/onboarding/SetupPage.test.tsx:125]: The hostname is actually a search param on the hash https://github.com/pixiebrix/pixiebrix-extension/blob/b5c30fd68bd3180fccf3c3360f79033e21b86821/src/background/installer.ts#L53-L53
```suggestion
      "chrome-extension://abc/options.html#/start?hostname=https://mycontrolroom.com"
```

Comment (twschiller) [reply src/options/pages/onboarding/SetupPage.test.tsx:125]: @mnholtz isn't this check wrong then? https://github.com/pixiebrix/pixiebrix-extension/blob/dda30f7c83ee1e04c90e8306a85613568fb6e977/src/options/pages/onboarding/partner/PartnerSetupCard.tsx#L104-L104

File: src/auth/token.ts [hunk 5]
```diff
@@ -145,12 +158,16 @@ export async function getExtensionAuth(): Promise<
...
  */
-export async function clearExtensionAuth(): Promise<void> {
+export async function clearCachedAuthSecrets(): Promise<void> {
   console.debug("Clearing extension auth");
-  await browser.storage.local.remove(STORAGE_EXTENSION_KEY);
-  await browser.storage.local.remove(STORAGE_PARTNER_TOKEN);
+  await Promise.all([
+    browser.storage.local.remove(STORAGE_EXTENSION_KEY),
+    browser.storage.local.remove(STORAGE_PARTNER_TOKEN),
+  ]);
```
Comment (fregante) [inline src/auth/token.ts:170]: .remove() accepts an array of keys to remove at once, we don't need to make two calls

**Pull request merged**